src/store/modules/emails.js

```diff
@@ -1,27 +1,95 @@
-const state = [
-  {
-    senderId: 1,
-    senderName: "Jerry",
-    senderIcon: "logo.png",
-    emails: [
+const state = {
+  selectedSenderId: 1,
+};
+
+// getters
+const getters = {
+  getEmails() {
+    //this would use the email service
+    let mail = [
```

Comment:
This object should be retrieved from state. Check out how chatMessages handles getChatMessageObject()